src/Sarif.WorkItemFiling/WorkItemFiler.cs

```diff
@@ -24,35 +31,64 @@ public class WorkItemFiler
         /// An object that implements the <see cref="IFileSystem"/> interface, providing
         /// access to the file system.
         /// </param>
-        public WorkItemFiler(IFileSystem fileSystem)
+        /// <param name="filingTarget">
+        /// An object that represents the system (for example, GitHub or Azure DevOps)
+        /// to which the work items will be filed.
+        /// </param>
+        public WorkItemFiler(FilingTarget filingTarget, IGroupingStrategy groupingStrategy, IFileSystem fileSystem)
         {
+            _filingTarget = filingTarget ?? throw new ArgumentNullException(nameof(filingTarget));
+            _groupingStrategy = groupingStrategy ?? throw new ArgumentNullException(nameof(groupingStrategy));
             _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
         }
 
         /// <summary>
         /// Files work items from the results in a SARIF log file.
         /// </summary>
-        /// <param name="path">
+        /// <param name="logFilePath">
         /// The path to the SARIF log file.
         /// </param>
-        public void FileWorkItems(string path)
+        /// <returns>
+        /// The set of results that were filed as work items.
+        /// </returns>
+        public async Task<IEnumerable<ResultGroup>> FileWorkItems(string logFilePath)
         {
-            if (path == null) { throw new ArgumentNullException(nameof(path)); }
+            if (logFilePath == null) { throw new ArgumentNullException(nameof(logFilePath)); }
+
+            string logFileContents = _fileSystem.ReadAllText(logFilePath);
+
+            EnsureValidSarifLogFile(logFileContents, logFilePath);
+
+            SarifLog sarifLog = JsonConvert.DeserializeObject<SarifLog>(logFileContents);
+
+            // CONSIDER: Multiple runs?
+            if (sarifLog.Runs[0]?.Results?.Count > 0)
```

Comment:
>sarifLog.Runs[0]? [](start = 16, length = 17)

i know this is a future todo based on the comment - we almost certainly would need to merge the sarifs into a single one before the bug filing step is executed. So, we will certainly need to iterate through multiple run objects.

Reply:
Ok, I will make that change now.

---
In reply to: [306062380](https://github.com/microsoft/sarif-sdk/pull/1580#discussion_r306062380) [](ancestors = 306062380)